functions/_fzf_search_history.fish

```diff
@@ -19,8 +19,11 @@ function _fzf_search_history --description "Search command history. Replace the
     )
 
     if test $status -eq 0
-        set command_selected (string split --max 1 " │ " $command_with_ts)[2]
-        commandline --replace -- $command_selected
+        set -U commands_selected # empty list to loop over selected commands, in case multiple
+        for c in (string split \n $command_with_ts)
```

Comment:
Commands can be multi-line. That's why the test broke. Thankfully, fish variables are arrays so you can just iterate through them. Can you try
```
for c in $command_with_ts)
```
Also can you rename the variable to `commands_with_ts`?